node/src/main.rs

```diff
@@ -561,11 +577,19 @@ async fn main() {
                 generic_store.clone(),
                 node_id.clone(),
             );
-            let subscription_server = GraphQLSubscriptionServer::new(
-                &logger,
-                graphql_runner.clone(),
-                generic_store.clone(),
-            );
+
+            // Disable the subscriptions server when running against a read-only
+            // database
+            if !readonly_db {
+                let subscription_server = GraphQLSubscriptionServer::new(
+                    &logger,
+                    graphql_runner.clone(),
+                    generic_store.clone(),
+                );
+
+                // Serve GraphQL subscriptions over WebSockets
+                graph::spawn(subscription_server.serve(ws_port));
```

Comment:
I'm a little uncomfortable with spawning subscriptions here potentially introducing a regression. Why was this spawned last before this commit? It's probably ok if there was any dependency on ordering before then there was already a bug because there are no synchronization primitives to indicate this dependency resolving. Just a concern.

Reply:
There shouldn't be any race conditions here, no. I think we just spawned all the servers at the bottom because it looked... clean.
![image](https://user-images.githubusercontent.com/19324/85594667-6b297100-b648-11ea-95c0-8a374360f6c2.png)